docs/api/memory/link_time_optimization.md

```diff
@@ -0,0 +1,66 @@
+# Link Time Optimization
+
+Link Time Optimization (LTO) is a program memory usage optimization mechanism that the compiler performs at link time. At compile time, the compiler creates a special intermediate representation of all translation units. It then optimizes them as a single unit at link time, which uses less memory than non-LTO builds.
+
+## Using LTO in Mbed OS
+
+The Mbed OS build system implements LTO as an optional profile extension in `tools\profiles\extensions\lto.json`. Enabling LTO amends the build profile with LTO flags. LTO performs heavy memory optimizations that break debugging, so we don't recommend enabling it in debug or develop profiles.
+
+To enable LTO, retype the `--profile` option with LTO file path `tools\profiles\extensions\lto.json`.
+
+<span class="notes">**Note**: For profile extensions you have to put the full path relative to the project's root folder.</span>
+
+To enable LTO with the `release` profile:
+
+```
+mbed compile -t TOOLCHAIN -m TARGET --profile release --profile mbed-os/tools/profiles/extensions/lto.json
+```
+
+Example LTO profile memory savings for [mbed-os-example-blinky](https://github.com/ARMmbed/mbed-os-example-blinky):
+
+|Build type|Total static RAM memory (data + BSS)|Total flash memory (text + data)|
+| --- | --- | --- |
+| GCC_ARM - release - no LTO | 12,096B | 44,628B |
```

Comment:
Do we expect these numbers to change? If so, we should remove the table, so we don't have to keep updating it.

Reply:
Frankly only amount of saved memory  is important. But true, it will change slightly depending on Mbed OS version used.

Maybe it would be better to round the numbers to kB like below example?
What do you think @jamesbeyond ?


|Build type|Total Static RAM memory (data + bss)|Total Flash memory (text + data)|
| --- | --- | --- |
| GCC_ARM - release - no LTO | 12.1 kB | 44.6 kB |
| GCC_ARM - release - LTO | 11.8 kB | 41.1 kB |
|***saved memory***| 0.3 kB| 3.5 kB|
| ARM - release - no LTO | 10.36 kB | 35.5 kB |
| ARM - release - LTO | 10.15 kB | 31.5 kB |
|***saved memory***| 0.21 kB|‭ 4 kB|